### DISCUSSION TITLE
Duplicate MAC addresses ; OUI db?

### DISCUSSION THREAD

**Discussion by bluikko** [netbox-community/netbox#7706] [Q&A] [answered]
While editing manually some devices I noticed that it is possible to add multiple (physical) interfaces with the same MAC address.

Is that intentional? Is there some use case for existence of duplicate MAC addresses?

I would assume that these days having same MAC address on multiple NICs would not happen anymore. I could see some benefit in having multiple MAC addresses in the database for virtual interfaces. Maybe the logic to detect duplicate MAC addresses would be too complex if they would be allowed only on some cases and having MAC address as a unique key would not be possible?

Edit:

I had a second point about MAC addresses. To me at least it would be beneficial to display the manufacturer based on an OUI database next to MAC addresses (maybe a column on its own on tables; or displayed as a read-only field next to MAC address when editing an Interface).

What do other people think about that?

**Comment by candlerb** [comment, marked as answer]
Duplicate MAC addresses can certainly happen legitimately, e.g. when you have a physical interface with multiple VLAN subinterfaces.

**Reply by bluikko** [reply]
Indeed. That is why I mentioned that virtual interfaces could have it.

But for physical interfaces? Or would the "duplicate detection logic" become too complex if it would need to differentiate between physical and logical/virtual interfaces?
In that case a simple unique key wouldn't probably work.

**Reply by jeremystretch** [reply]
Yes, it's quite common for multiple physical interfaces on a device to share the same MAC address, which is why NetBox support this.

**Comment by lowinger42** [comment]
Den 2021-11-01 kl. 09:58, skrev bluikko:
> Indeed. That is why I mentioned that virtual interfaces could have it.

As Jeremy pointed out, it is fairly common. A MAC address only needs to 
be unique per L2 domain.

 From an Arista:


show interfaces | i (^Ethernet.* | Hardware is Ethernet, address is)

Ethernet1 is up, line protocol is up (connected)
   Hardware is Ethernet, address is 985d.82b7.771b
Ethernet2 is up, line protocol is up (connected)
   Hardware is Ethernet, address is 985d.82b7.771b
Ethernet3 is up, line protocol is up (connected)
   Hardware is Ethernet, address is 985d.82b7.771b

Ethernet4 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.771f (bia 985d.82b7.771f)
Ethernet5 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.7720 (bia 985d.82b7.7720)
Ethernet6 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.7721 (bia 985d.82b7.7721)
Ethernet7 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.7722 (bia 985d.82b7.7722)
Ethernet8 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.7723 (bia 985d.82b7.7723)
Ethernet9 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.7724 (bia 985d.82b7.7724)
Ethernet10 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.7725 (bia 985d.82b7.7725)
Ethernet11 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.7726 (bia 985d.82b7.7726)
Ethernet12 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.7727 (bia 985d.82b7.7727)
Ethernet13 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.7728 (bia 985d.82b7.7728)
Ethernet14 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.7729 (bia 985d.82b7.7729)
Ethernet15 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.772a (bia 985d.82b7.772a)


Ports with different mac addresses are switchports.

For example, reconfiguring Ethernet4 with "no switchport" changes the 
MAC address.

Ethernet4 is down, line protocol is notpresent (notconnect)
   Hardware is Ethernet, address is 985d.82b7.771b



-- 
MVH / Regards
Anders Löwinger

**Reply by bluikko** [reply]
For "layer 2 ports" that makes sense. The MAC address you see on such ports is likely not shared with L3 ports.
